doc/howtos/OSTree.rst

```diff
@@ -0,0 +1,127 @@
+Handling System Updates Using OSTree
+####################################
+
+IoT Reference OS Kit ('refkit') has support for updating devices running
+one of the refkit profile images using an HTTP/HTTPs server and OSTree.
+Ostree is a library and suite of command line tools, largely inspired by
+git, that provides version control and a distribution mechanism for bootable
+OS filesystem trees, or other binaries. For a comprehensive introduction
+to OSTree, an overview of its architecture and feature set, please refer
+to its `documentation <http://ostree.readthedocs.io>_`.
+
+If enabled, OSTree support in refkit
+
+ * provides A/B-versioning (within a single block device partition)
+ * creates a per-image OSTree repository for each image built
+ * exposes builds in a common OSTree repository as a series of commits
+ * can sign each commit (which is then verified during updates)
+ * can provide a service for automatic image updates
+
+
+Enabling OSTree Support
+#######################
+
+To enable end-to-end OSTree support with automatic image updates, you will
+need to
+
+ * enable OSTree support for your builds
+ * ideally generate and use a pair of signing keys for your builds/updates
+ * expose the OSTree repository of your builds over HTTP/HTTPS
+ * point your clients to the exposed OSTree repository
+
+To enable OSTree support, turn on the 'ostree' image feature. You can
+do this by incuding the following configuration snippet in your local.conf
+or other suitable global Yocto configuration file::
+
+    REFKIT_IMAGE_EXTRA_FEATURES += "ostree"
+
+To use your GPG signing key pair for signing and verifying the OSTree
+repository commits, assuming your keys are in the keyrings in <gpg-home>
+with key ID release@example.org, add the following to your local.conf or
+other suitable global Yocto configuration file::
+
+    OSTREE_GPGDIR = "<gpg-home>"
+    OSTREE_GPGID  = "release@example.org"
```

Comment:
I think it's worthwhile to add a note here that this is not necessary when building refkit images in the default development mode and why, and to explain the implications (image update not secure).